app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -361,13 +369,19 @@ def rbac_tenant_delete
       if params[:id].nil? || Tenant.find_by_id(params[:id]).nil?
         add_flash(_("%s no longer exists") % "Tenant", :error)
       else
-        tenants.push(params[:id])
+        if Tenant.find_by_id(params[:id]).divisible
```

Comment:
you are hitting the database 3 times for the same tenant. And if you are in the `else` block, `params[:id]` is not `nil`.

```ruby
else # showing 1 tenant, delete it
  if (tenant = Tenant.find_by(:id => params[:id]))
    if tenant.divisible
      tenants.push(params[:id])
    else
      projects.push(params[:id])
    end
    parent_id = tenant.parent.id
    self.x_node = "tn-#{to_cid(parent_id)}"
  else
    add_flash(_("%s no longer exists") % "Tenant", :error)
  end
end
```